crates/symbolicator/src/services/symcaches/markers.rs

```diff
@@ -0,0 +1,147 @@
+use std::collections::HashSet;
+use std::io::SeekFrom;
+
+use crate::services::bitcode::BcSymbolMapHandle;
+
+/// This is the legacy marker that was used previously to flag a SymCache that was created
+/// using a`BcSymbolMap`.
+const SYMBOLMAP_MARKER: &[u8] = b"WITH_SYMBOLMAP";
+
+/// This marker denotes that a SymCache has appended markers.
+const MARKERS_MARKER: &[u8] = b"WITH_MARKERS";
+
+/// Encapsulation of all the source artifacts that are being used to create SymCaches.
+#[derive(Clone, Debug, Default)]
+pub struct SecondarySymCacheSources {
+    pub bcsymbolmap_handle: Option<BcSymbolMapHandle>,
+}
+
+/// This is the markers that are being embedded into, and read from, a SymCache file.
+#[derive(Debug, Default, PartialEq)]
+pub struct SymCacheMarkers {
+    markers: HashSet<String>,
```

Comment:
I think it'd be better to use something stronger typed rather than "string".  Can this not be an enum?